### DISCUSSION TITLE
Option with collection properties

### DISCUSSION THREAD

**Discussion by rezanid** [VsixCommunity/Community.VisualStudio.Toolkit#511] [Q&A]
# The Intro
First of all, thanks a lot for the great work you are doing here! I have started with Vsix development a while back and currently working on a new package that is going to provide a comprehensive tooling for Power Platform developers. It's still fairly new. Here is the link the repo: https://github.com/rezanid/xrmtools the code is not well organized at this stage as it started as a PoC.

# The question!
Here there is a nice [recipe to build OptionPages](https://www.vsixcookbook.com/recipes/settings-and-options.html), but I was wondering if having an IEumerable<Something> as a property of my options is even supported? Something like this:

```csharp
public class General : BaseOptionModel<General>
{
    [Category("Dataverse")]
    [DisplayName("Environments")]
    [Description("List of Power Platform environments.")]
    [DefaultValue(true)]
    public List<DataverseEnvironment> Environments { get; set; } = [];
}

public class DataverseEnvironment
{
    public string Name { get; set; }
    public string Url { get; set; }
    public string ConnectionString { get; set; }
}
```

# A little more context
The idea is devs can add as many Power Platform environments as they want and then either select one as the active environment or select an environment per each project. I am still in two minds about the second part.

Separating the list of environments from the active environments allow devs to take their catalog with them anywhere, yet in each project select which one to use. I would also appreciate any ideas / comments one this part.

More information can be found in this issue: https://github.com/rezanid/xrmtools/issues/2

**Comment by madskristensen** [comment]
Here's how to do enums: https://github.com/madskristensen/SolutionColors/blob/master/src/Options/General.cs#L63

**Reply by rezanid** [reply]
Thanks, I've seen that one. But, I was wondering if it's a supported scenario for the option to have a property of List<T> or IEnumerable<T> or any of the other collection types. If not, what would be the alternative?

**Reply by rezanid** [reply]
Never mind. It simply worked. I was just looking in the wrong place! Thanks again.

**Comment by rezanid** [comment]
Collection properties are already supported and by default Visual Studio displays CollectionEditor similar to when you have a component in the toolbox in Windows form editor and want to edit a collection property.
